### PR TITLE
System check for multiple l10n dirs

### PR DESCRIPTION
**Pull request by demeritcowboy**
Overview
----------------------------------------
After https://github.com/civicrm/civicrm-core/pull/28061 ~~old~~ long-time administrators will still be doing what they normally do and putting the translation files either under civicrm_root or somewhere else. Depending on how exactly they are set up, after install with either CLI or UI a duplicate copy of the files may get downloaded into a different location and civi will get set to use those instead. But then later the administrator will update their original translation files. Head-scratching follows when the site doesn't seem to be using the new translations.

A typical example might be the admin puts the files in sites/all/modules/civicrm/l10n for drupal 7. If the admin does a stock install in a non-en_US language without any further configuration the installer will put another copy of the language files in sites/default/files/civicrm/l10n.

Before
----------------------------------------
Confusion.

After
----------------------------------------

Technical Details
----------------------------------------

Comments
----------------------------------------
You don't have to do a full install to at least partially test this. You could take a stock install even in en_US and just create an l10n folder at sites/all/modules/civicrm/l10 and another at sites/default/files/civicrm/l10.

### CODE REVIEW

File: CRM/Utils/Check/Component/Env.php [hunk 1]
```diff
@@ -1127,4 +1127,50 @@ public function checkAngularModuleSettings() {
...
   }
 
+  public function checkForMultipleL10NDirs() {
+    $messages = [];
+    // This is what civi thinks is the current l10n path.
+    // Even if the site is currently en_US, we still want to do this check
+    // since they could change the language later.
+    $current_l10n = self::normalizePath(\Civi::Paths()->getPath('[civicrm.l10n]/'));
+    // use array keys instead of values to automatically dedupe paths
+    $dirs = [$current_l10n => 1];
```
Comment (totten) [inline CRM/Utils/Check/Component/Env.php:1137]: Should `$current_l10n` have the same kind of guard as the others? 

```php
if (is_dir($current_l10n)) { $dirs[$current_l10n] = 1; }
```

If it doesn't exist, then I guess it's not party to a conflict... although it is different kind of error...

Comment (demeritcowboy) [reply CRM/Utils/Check/Component/Env.php]: I was waffling on that. I was maybe trying to do 2 checks in one but it's not that useful.

**Comment by totten** [comment]
I tried it out locally and shuffled around my `l10n` folders to see how it behaved. The warning came on+off as expected.

Test failure is unrelated.

Merging.

**Comment by mlutfy** [comment]
Thank you @demeritcowboy @totten :)

**Pull request merged**